server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java

```diff
@@ -951,7 +952,7 @@ public void testMinScore() throws Exception {
         context.setSize(1);
         context.trackTotalHitsUpTo(5);
 
-        QueryPhase.executeInternal(context);
+        QueryPhase.addCollectorsAndSearch(context);
         assertEquals(10, context.queryResult().topDocs().topDocs.totalHits.value);
 
         reader.close();
```

Comment:
Do you think it would be possible to add some unit tests to this class (QueryPhaseTests) around the QueryPhase changes you made? For instance, we could test the case where the initial search for aggs is or isn't performed depending on track_total_hits and aggs in the request. We could test the execution of the following queries that are kept separate etc.

Reply:
Do we need tests also for named queries and stats ( I figured that's what the node queue size and service time are used for)?

Reply:
Added a test to check if queries are executed in the expected order.

Any suggestions for stats?

Reply:
I'd try and verify that when the stats parameter is provided in combination with RRF the behaviour is the expected one. We could do this in a unit test around the actual execute method in QueryPhase, as well as an integration test that verifies the end-to-end behaviour.

Reply:
it's fine to leave this for a follow-up